gcp_variant_transforms/libs/partitioning.py

```diff
@@ -270,6 +314,64 @@ def copy_to_flatten_table(self, output_base_table_id):
       logging.info('Flatten table is fully loaded: %s', full_output_table_id)
 
 
+class ParseTableSchema(object):
+  """Parses json file containing BigQuery table schema adds description to it"""
+
+  def __init__(self, schema_dict, description_dict, flattened_col=''):
+    # type: (dict, dict, str) -> None
+    """Copies column descriptions from variant table to sample table schema.
+
+    Because of the way we create sample optimized tables schema, we miss the
+    column descriptions:
+      * Compose a query to flatten call column.
+      * Write the output of the query (LIMIT 1) to a temporary table.
+      * Extract the temporary table's schema into a json file.
+
+    This class enables use to add 'description' fields to the json schema file.
+    Description values are given via `description_dict` that is build by
+    querying INFORMATION_SCHEMA.COLUMN_FIELD_PATHS meta-table.
+
+    Args:
+      schema_dict: A dict that is output of json.load() of schema file.
+      description_dict: {field_name : description} dictionary.
+      flattened_col: A string to identify the flatten column. For example, we
+                     must know `call_sample_id` is the same as `call.sample_ld`.
+
+    Returns:
+      None, modifies the input `schema_dict` and add `description` fields to it.
+    """
+    self._schema_dict = schema_dict
+    self._description_dict = description_dict
+    if flattened_col:
+      self._is_schema_flatten = True
+      self._flattened_col_prefix = flattened_col + '_'
+      self._flattened_col_path = flattened_col + '.'
+    else:
+      self._is_schema_flatten = False
+
+  def _get_name(self, item):
+    field_name = item.get('name')
+    if self._is_schema_flatten:
+      field_name = field_name.replace(self._flattened_col_prefix,
+                                      self._flattened_col_path, 1)
+    return field_name
+
+  def _parse_schema(self, path, schema_dict):
+    for item in schema_dict:
```

Comment:
Will this change the order of the fields in the scheme? If so, can it affect the vcf-to-bq-to-vcf pipeline (ie, original input not matching output)?

Reply:
No, we don't modify the oder or items in the list, we only add to each item `item['description']` field.